crates/oxc_isolated_declarations/src/lib.rs

```diff
@@ -405,6 +405,67 @@ impl<'a> IsolatedDeclarations<'a> {
     }
 
     pub fn report_error_for_expando_function(&self, stmts: &oxc_allocator::Vec<'a, Statement<'a>>) {
+        let mut assignable_properties_for_namespace = FxHashMap::<&str, FxHashSet<Atom>>::default();
+        for stmt in stmts {
```

Comment:
Looks like we don't need to iterate over `stmts` again. We have an iteration for collecting `can_expando_function_names`

Reply:
In that iteration we also report the error. Since namespace declarations can come after the function declaration, we need to ensure that we've iterated over all statements first. Example:
```ts
export function foo(): void {}
foo.a = 42;  // When we get here we'll report an error because we did not see the namespace declaration yet.
export namespace foo {
    export let a: number;
}
```

Reply:
I see, Thanks